pkg/completion/completion_test.go

```diff
@@ -190,6 +190,40 @@ func (s *CompletionSuite) TestCompletionWithCallback(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *CompletionSuite) TestCompletionWithRetryCallback(c *C) {
+	var err error
+	var callbackCount, retryCount int
+
+	ctx, cancel := context.WithTimeout(context.Background(), TestTimeout)
+	defer cancel()
+
+	// Set a shorter timeout to shorten the test duration.
+	wg := NewWaitGroup(ctx)
+
+	comp := wg.AddCompletionWithCallbacks(func() { callbackCount++ }, func() { retryCount++ })
+	c.Assert(comp.Context(), Equals, ctx)
+
+	// Retry is not idempotent.
+	comp.Retry()
+	comp.Retry()
+	comp.Retry()
+
+	// Complete is idempotent.
+	comp.Complete()
+	comp.Complete()
+	comp.Complete()
+
```

Comment:
I would also call `comp.Retry()` here again, to show that it has no effect after completion.

Reply:
OK, somehow I thought I did that already...

Reply:
Added